data_structures/binary_trees/words_alphabetical.c

```diff
@@ -0,0 +1,309 @@
+ * another file "wordcount.txt"
+ * @details
+ * Given a file (`file.txt`) containing words (like a publication or a novel),
+ * where words are separated by a space, newline, or underscore.
+ * This program prints (writes or outputs) to another file (`wordcount.txt`),
+ * the individual words contained in 'file.txt' with their frequencies (number
+ * of occurences) each on a newline and in alphabetical order. This program uses
+ * the binary tree data structure to accomplish this task.
+ * @author [Randy Kwalar](https://github.com/RandyKdev)
+ */
+
+#include <assert.h>    /// for assert
+#include <ctype.h>     /// for type checks
+#include <inttypes.h>  /// for uint64_t based types, int64_t based types
+#include <stdbool.h>   /// for boolean data type
+#include <stdio.h>     /// for IO operations
+#include <stdlib.h>    /// for memory allocation
+#include <string.h>    /// for string operations
+
+/**
+ * @brief structure defining a node in the binary tree
+ */
+struct Node
+{
+    char *word;          ///< the word (value) of the node
+    uint64_t frequency;  ///< number of occurences of the word
+    struct Node *left;   ///< pointer to the left child node
+    struct Node *right;  ///< pointer to the right child node
+};
+
+/**
+ * @brief Ends program due to an error
+ * @param errorMessage the error message to be printed
+ * @returns void
+ */
+void endProgramAbruptly(char *errorMessage)
+{
+    fprintf(stderr, "%s\n", errorMessage);
+    exit(EXIT_FAILURE);
+}
+
+/**
+ * @brief Frees memory when program is terminating
+ * @param node pointer to current node
+ * @returns void
+ */
+void freeTreeMemory(struct Node *node)
+{
+    if (node != NULL)
+    {
+        freeTreeMemory(node->left);
+        freeTreeMemory(node->right);
+        free(node->word);  // freeing node->word because memory was allocated
+                           // using malloc
+        free(node);  // freeing node because memory was allocated using malloc
+    }
+}
+
+/**
+ * @brief Stores word in memory
+ * @param word word to be stored in memory
+ * @returns a pointer to the newly allocated word
+ */
+char *getPointerToWord(char *word)
+{
+    char *string =
+        (char *)malloc((strlen(word) + 1) * sizeof(char));  // pointer to string
+    // + 1 is for the '\0' character
+    if (string != NULL)
+    {
+        strcpy(string, word);
+        return string;
+    }
+    endProgramAbruptly(
+        "\nA problem occurred while reserving memory for the word\n");
+    return NULL;
+}
+
+/**
+ * @brief Closes file after read or write
+ * @param file pointer to file to be closed
+ * @returns void
+ */
+void closeFile(FILE *file)
+{
+    if (fclose(file))
+        endProgramAbruptly("\nA Problem Occurred while closing a file\n");
+}
+
+/**
+ * @brief Reserves memory for new node
+ * @returns a pointer to the newly allocated node
+ */
+struct Node *allocateMemoryForNode()
+{
+    struct Node *node =
+        (struct Node *)malloc(sizeof(struct Node));  // pointer to node
```

Comment:
```suggestion
        (struct Node *)malloc(sizeof(struct Node));  ///< pointer to the node
```

Reply:
> Try not to accept the suggestions so quickly. Instead, accept one suggestion, wait a minute, and accept the next suggestion. There could be problems with the GitHub workflow removing those changes if you don't do it that way.

Ok, got it.
Would like to find out whether problems will arise with the Github workflow when suggestions are grouped and committed as a batch?